### PR TITLE
[luigi.contrib.spark] pyspark python options added

### PR DESCRIPTION
**Pull request by drowoseque**
## Description
Added `hadoop_user_name`, `pyspark_python`, `pyspark_driver_python` options for SparkSubmitTask

## Motivation and Context
It's a usual usecase when you want to submit your spark application with a specific python environment on spark workers or under specific user

## Have you tested this? If so, how?
I have included a unit test for that
That approach works under my production environment

### CODE REVIEW

**Comment by drowoseque** [comment]
@honnix  @Tarrasch @dlstadther  could you please review this PR?

**Comment by drowoseque** [comment]
@mrk-its 
@GoodDok 
could you review this PR please ?

**Comment by drowoseque** [comment]
@GoodDok
 your turn dude :)

File: luigi/contrib/spark.py [hunk 1]
```diff
@@ -168,15 +192,13 @@ def archives(self):
     def hadoop_conf_dir(self):
         return configuration.get_config().get(self.spark_version, "hadoop-conf-dir", None)
 
-    def get_environment(self):
```
Comment (honnix) [inline luigi/contrib/spark.py]: This seems to be a breaking change though.

Comment (drowoseque) [reply luigi/contrib/spark.py]: @honnix 
it's an internal method used by `SparkSubmitTask` in `program_environment` only. 
But I agree that it's better not to change

Comment (drowoseque) [reply luigi/contrib/spark.py]: @honnix 
done

Comment (honnix) [reply luigi/contrib/spark.py]: Thanks. It's unfortunate we leaked this method as 'public'.

**Comment by mfcabrera** [comment]
I wanted to say that this was already possible using the `conf` property of the Task. 

```python
 @property
    def conf(self):
        conf = super().conf or {}

        conf["spark.pyspark.driver.python"] = "bar"
        conf["spark.pyspark.python"] =  "foo"
        return conf
``` 

I prefer this way as I find it more flexible but I don't think having those additional properties hurt though. 

Cheers :).

**Pull request merged**